### PR TITLE
clean up json handling

### PR DESCRIPTION
**Pull request by macklin-10x**
Replaces all custom JSON parsing code with serde.

### CODE REVIEW

File: enclone_exec/tests/enclone_test1.rs [hunk 1]
```diff
@@ -647,99 +658,70 @@ fn test_licenses() {
...
+        .sorted()
+        .collect();
+
+    assert!(
+        fails.is_empty(),
+        "\nLicense check failed.  The following packages had problems:\n{}\n\
+        You may want to retry the test, since the license checks fails sporadically \
+        at a low rate.\n",
+        fails.join(", "),
+    );
```
Comment (sreenathkrishnan) [inline enclone_exec/tests/enclone_test1.rs:726]: we could delegate the license check to `cargo-deny`?

File: enclone_tools/src/bin/filtered_from_json.rs [hunk 2]
```diff
@@ -41,59 +33,48 @@ fn main() {
...
-            }
-            if ann[j]["feature"]["region_type"] == "D-REGION" {
-                d_gene = strip(&ann[j]["feature"]["gene_name"].to_string());
-            }
-            if ann[j]["feature"]["region_type"] == "J-REGION" {
-                j_gene = strip(&ann[j]["feature"]["gene_name"].to_string());
-            }
-            if ann[j]["feature"]["region_type"] == "C-REGION" {
-                c_gene = strip(&ann[j]["feature"]["gene_name"].to_string());
+        for region in ann.annotations {
```
Comment (sreenathkrishnan) [inline enclone_tools/src/bin/filtered_from_json.rs:55]: We could use this function: 
https://github.com/10XGenomics/enclone_ranger/blob/main/vdj_ann/src/annotate.rs#L3235
```
let v_gene = ann.get_gene_name(VdjRegion::V).cloned().unwrap_or_default();
```

File: enclone_tools/src/bin/filtered_from_json.rs [hunk 3]
```diff
@@ -1,36 +1,28 @@
...
+use vdj_ann::annotate::ContigAnnotation;
+use vdj_types::VdjRegion;
 
-fn strip(x: &str) -> String {
-    if x == "null" {
-        String::new()
-    } else {
-        x.between("\"", "\"").to_string()
-    }
+// FIXME duplicated between here and enclone_main/src/subset.rs
```
Comment (sreenathkrishnan) [inline enclone_tools/src/bin/filtered_from_json.rs:11]: Addressing in this PR?

**Review by sreenathkrishnan — APPROVED** [review]
Much nicer! Glad that we got rid of custom json writing and are using serde json

File: enclone_exec/tests/enclone_test1.rs (hunk 1 above)
Comment (macklin-10x) [reply enclone_exec/tests/enclone_test1.rs:726]: Absolutely and I can address this in a follow-up PR.

File: enclone_tools/src/bin/filtered_from_json.rs (hunk 3 above)
Comment (macklin-10x) [reply enclone_tools/src/bin/filtered_from_json.rs:11]: Yeah I can clean this up.

File: enclone_tools/src/bin/filtered_from_json.rs (hunk 2 above)
Comment (macklin-10x) [reply enclone_tools/src/bin/filtered_from_json.rs:55]: Done.

File: enclone_tools/src/bin/filtered_from_json.rs (hunk 3 above)
Comment (macklin-10x) [reply enclone_tools/src/bin/filtered_from_json.rs]: Done.

**Pull request merged**